app/actions/views/channel.js

```diff
@@ -200,18 +202,54 @@ export function selectDefaultChannel(teamId) {
         const channel = selectChannelByName(channelsInTeam, getRedirectChannelNameForTeam(state, teamId));
         let channelId;
         if (channel) {
-            channelId = channel.id;
-        } else {
-            // Handle case when the default channel cannot be found
-            // so we need to get the first available channel of the team
-            const channels = Object.values(channelsInTeam);
-            const firstChannel = channels.length ? channels[0].id : '';
-            channelId = firstChannel;
+            dispatch(handleSelectChannel(channel.id));
+            return;
+        }
+
+        // Handle case when the default channel cannot be found
+        // so we need to get the first available channel of the team
+        const channels = Object.values(channelsInTeam);
+        const firstChannel = channels.length ? channels[0].id : '';
+        channelId = firstChannel;
+
+        if (channelId) {
+            dispatch(handleSelectChannel(channelId));
+            return;
+        }
+
+        // Handle case when no channel found in the current team
+        // so we need to get the first available default channel of all teams
+        const teamsDefaultChannels = getDefaultChannelForTeams(state);
+        for (const ch in Object.values(teamsDefaultChannels)) {
+            if (ch) {
+                channelId = ch.id;
+                break;
+            }
         }
 
         if (channelId) {
             dispatch(handleSelectChannel(channelId));
+            return;
+        }
+
+        // Handle case when no default channel found in any team
+        // so we need to get the first available channel of all teams
+        const teamsMyFirstChannels = getMyFirstChannelForTeams(state);
+        for (const ch in Object.values(teamsMyFirstChannels)) {
+            if (ch) {
+                channelId = ch.id;
+                break;
+            }
         }
+
+        if (channelId) {
+            dispatch(handleSelectChannel(channelId));
```

Comment:
don't you need to select the right team here too ?

Reply:
I don't think so, it is handled by the `handleSelectChannel` action it emits a `SELECT_CHANNEL` that includes the `teamId` extracted from the `channel` object itself.

Reply:
What? HandleSelectChannel does not select a team

Reply:
Ok, the handle selectChannel was passing the `team_id` parameter in the extra field of the action, and I was expecting was used to set the team too.. but yes, I think we should be selecting the team to there.

Reply:
Fixed here f8d229710f47b97a68a9fef56ddce22fdd2a276d